addons/gamification/models/res_users.py

```diff
@@ -367,3 +374,13 @@ def action_karma_report(self):
                 'search_default_user_id': self.id,
             },
         }
+
+    def _update_last_interacted(self):
+        """Log that a non-public user just interacted with the application.
+
+        Does nothing if on a public user or if the user doesn't yet have any login record.
+        """
+        self.ensure_one()
+        if self._is_public() or not self.login_date:  # no res_users_log record
+            return
+        self.last_interacted_date = fields.Datetime.now()
```

Comment:
even if it bypass freezetime it's still possible to mock it (and it's a better practise to use env.cr.now)
```py
with freeze_time(datetime), patch.object(self.env.cr, 'now', lambda: datetime):
    ...
```

Reply:
You're right. My problem was with the things written in db by postgres and the fact that datetime checks in the goals sql query were strict. By using `<=` instead of `<`, the test works fine too, and I can `use Cursor.now()`.
Also, since we dropped the time based-cache, I can avoid freezing time altogether.